osu.Game.Rulesets.Mania/Difficulty/ManiaDifficultyAttributes.cs

```diff
@@ -34,6 +53,9 @@ public override void FromDatabaseAttributes(IReadOnlyDictionary<int, double> val
 
             StarRating = values[ATTRIB_ID_DIFFICULTY];
             GreatHitWindow = values[ATTRIB_ID_GREAT_HIT_WINDOW];
+            OverallDifficulty = values[ATTRIB_ID_OVERALL_DIFFICULTY];
+            NoteCount = (int)values[ATTRIB_ID_NOTE_COUNT];
+            HoldNoteCount = (int)values[ATTRIB_ID_HOLD_NOTE_COUNT];
```

Comment:
I was going to ask why these are not just being read from the `IBeatmapOnlineInfo` param here and skipped entirely in `.ToDatabaseAttributes()`, but then realised that wouldn't work because of keymods. I'm not sure how intentional this was but yeah.

This is going to be something that we need to be wary of going forward unless realtime calculation becomes a thing.

Reply:
This part in particular was added by smoogi to calculate a sheet for the rework